src/main/java/com/uwaterloo/connect/serviceImpl/ActivityServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+package com.uwaterloo.connect.serviceImpl;
+
+import com.uwaterloo.connect.model.Activity;
+import com.uwaterloo.connect.model.ActivityRequest;
+import com.uwaterloo.connect.model.Attachment;
+import com.uwaterloo.connect.model.Post;
+import com.uwaterloo.connect.repository.ActivityRepository;
+import com.uwaterloo.connect.repository.AttachmentRepository;
+import com.uwaterloo.connect.repository.PostRepository;
+import com.uwaterloo.connect.service.ActivityService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.uwaterloo.connect.Constants.Constants.ERROR;
+import static com.uwaterloo.connect.Constants.Constants.SUCCESS;
+
+@Service
+public class ActivityServiceImpl implements ActivityService {
+
+    @Autowired
+    ActivityRepository activityRepository;
+
+    @Autowired
+    PostRepository postRepository;
+
+    @Autowired
+    AttachmentRepository attachmentRepository;
+
+    @Override
+    public String createActivity(ActivityRequest activityRequest){
+        try{
+
+            //New post
+            Integer postId = createPostForActivity(activityRequest);
+            //Save attachments
+            createAttachmentsforPost(postId, activityRequest.getFiles());
+            System.out.println("act: "+postId);
+            //Make activity for the new post
+            Activity activity = new Activity();
+            activity.setPostId(postId);
+            activity.setCategoryId(activityRequest.getCategoryId());
+            activity.setStatusId(activityRequest.getStatusId());
+            activity.setRecurring(activityRequest.isRecurring());
+            // Save dateTime
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
+            activity.setStartTime(LocalDateTime.parse(activityRequest.getStartTime(), formatter));
+            activity.setEndTime(LocalDateTime.parse(activityRequest.getEndTime(), formatter));
+            //Save Activity
+            activityRepository.save(activity);
+
+            return SUCCESS;
+        }catch (Exception e){
+            return ERROR+e;
+        }
+    }
+    public Integer createPostForActivity(ActivityRequest activityRequest){
+        try{
+            Post post = new Post(activityRequest.getUserId(), activityRequest.getPostText(), activityRequest.isShared());
```

Comment:
There really should be a single point in our code where a post is created. This will create issues in future if we change the creation logic in any way.

Reply:
I don't think we should call a controller method from inside our code, it should be from browser only.